app/(gcforms)/[locale]/(form administration)/forms/[[...path]]/components/client/Menu.tsx

```diff
@@ -0,0 +1,125 @@
+"use client";
+import {
+  MenuDropdown,
+  MenuDropdownItemI,
+  MenuDropdownItemCallback,
+} from "@clientComponents/myforms/MenuDropdown/MenuDropdown";
+import { useTranslation } from "react-i18next";
+import copy from "copy-to-clipboard";
+import { getForm } from "../../actions";
+import { getDate, slugify } from "@lib/client/clientHelpers";
+import { useCallback, useState } from "react";
+import { ConfirmDelete } from "./ConfirmDelete";
+
+export const Menu = ({
+  id,
+  name,
+  isPublished,
+  direction = "up",
+}: {
+  id: string;
+  name: string;
+  isPublished: boolean;
+  direction?: "up" | "down";
+}) => {
+  const {
+    t,
+    i18n: { language },
+  } = useTranslation(["my-forms"]);
+  const [showConfirm, setShowConfirm] = useState<boolean>(false);
+
+  const handleDelete = useCallback(() => {
+    setShowConfirm(true);
+  }, []);
+
+  const menuItemsList: Array<MenuDropdownItemI> = [
+    {
+      title: t("card.menu.save"),
+      callback: () => {
+        downloadForm(name, id);
+        return { message: "" };
+      },
+    },
+    {
+      title: t("card.menu.settings"),
+      url: `/${language}/form-builder/settings/${id}`,
+    },
+    {
+      title: t("card.menu.delete"),
+      callback: () => {
+        handleDelete();
+        return {
+          message: "",
+        };
+      },
+    },
+  ];
+
+  // Show slightly different list items depending on whether a Published or Draft card
+  if (isPublished) {
+    menuItemsList.unshift({
+      title: t("card.menu.copyLink"),
+      callback: copyLinkCallback,
+    });
+  } else {
+    menuItemsList.unshift(
+      {
+        title: t("card.menu.edit"),
+        url: `/${language}/form-builder/${id}/edit`,
+      },
+      {
+        title: t("card.menu.preview"),
+        url: `/${language}/form-builder/preview/${id} `,
```

Comment:
The form ID should be before the route name

```suggestion
        url: `/${language}/form-builder/${id}/preview `,
```